ext/windows/service/daemon.rb

```diff
@@ -155,12 +155,18 @@ def report_windows_event(type, id, message)
     end
   end
 
+  # Parses runinterval.
+  #
+  # @param puppet_path [String] The file path for the Puppet executable.
+  # @return runinterval [Integer] How often to do a Puppet run, in seconds.
   def parse_runinterval(puppet_path)
     begin
-      runinterval = %x(#{puppet_path} config --section agent --log_level notice print runinterval).to_i
-      if runinterval == 0
+      runinterval = %x(#{puppet_path} config --section agent --log_level notice print runinterval)
+      if runinterval == ''
         runinterval = 1800
         log_err("Failed to determine runinterval, defaulting to #{runinterval} seconds")
+      else
+        runinterval = runinterval.to_i
```

Comment:
Wondering if we should be more defensive as `String#to_i` has some surprising behavior:

```
irb(main):002:0> "foo".to_i
=> 0
```

However, this seems more strict/correct:

```
irb(main):003:0> Integer("foo")
(irb):3:in `Integer': invalid value for Integer(): "foo" (ArgumentError)
        from (irb):3:in `<main>'    
```

In theory, this should never happen, because `puppet config print runtinterval` doesn't allow invalid durations to be set:

```
$ bundle exec puppet config set runinterval minutes --section main
Error: Invalid duration format '"minutes"' for parameter: runinterval
Error: Try 'puppet help config set' for usage
```
And if something else writes to `puppet.conf`, the `puppet config print` command will error and return empty string to stdout:

```
$ grep runinterval ~/.puppetlabs/etc/puppet/puppet.conf 
runinterval=minutes
$ bundle exec puppet config print runinterval --section main
Error: Invalid duration format '"minutes"' for parameter: runinterval
Error: Try 'puppet help config print' for usage
```

That said, 0 means run all the time, so I think additional strictness would be good.